### PR TITLE
[MI-2209] Added slash command for repos

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
1. Added a command to open the subscription modal. 
2. Added a command to list all the repo subscriptions. 
3. Added a command to delete a repo subscription.

### CODE REVIEW

**Review by ayusht2810 — COMMENTED** [review]
why there are some commits not made by you in this PR?

**Comment by raghavaggarwal2308** [comment]
@ayusht2810 I took a pull from `MI-2193` because of some changes in the slash command by Abhishek. I think that's why it shows Abhishek's commits on that branch here.

File: server/plugin/api_test.go [hunk 1]
```diff
@@ -590,6 +592,7 @@ func TestHandleCreateSubscriptions(t *testing.T) {
 				"organization": "mockOrganization",
 				"project": "mockProject",
 				"eventType": "mockEventType",
+				"serviceType":"mockServiceType",
```
Comment (ayusht2810) [inline server/plugin/api_test.go:595]:
```suggestion
				"serviceType": "mockServiceType",
```

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -544,6 +544,7 @@ func TestHandleCreateSubscriptions(t *testing.T) {
...
```
Comment (ayusht2810) [inline server/plugin/api_test.go:547]:
```suggestion
				"serviceType": "mockServiceType",
```

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/command.go [hunk 3]
```diff
@@ -176,15 +208,15 @@ func azureDevopsUnsubscribeCommand(p *Plugin, c *plugin.Context, commandArgs *mo
...
 	}
 
-	return p.sendEphemeralPostForCommand(commandArgs, fmt.Sprintf("Boards subscription with ID: %q does not exist", subscriptionIDToBeDeleted))
+	return p.sendEphemeralPostForCommand(commandArgs, fmt.Sprintf("%s subscription with ID: %q does not exist", cases.Title(language.Und).String(command), subscriptionIDToBeDeleted))
 }
 
-func azureDevopsListSubscriptionsCommand(p *Plugin, c *plugin.Context, commandArgs *model.CommandArgs, args ...string) (*model.CommandResponse, *model.AppError) {
-	// If 4th argument is present then it must be "all_channels"
+func azureDevopsListSubscriptionsCommand(p *Plugin, c *plugin.Context, commandArgs *model.CommandArgs, command string, args ...string) (*model.CommandResponse, *model.AppError) {
+	// If 4th argument is present then it must be "all-channels"
```
Comment (avas27JTG) [inline server/plugin/command.go:219]: why this change? `all_channels` is the command param

File: server/plugin/utils.go [hunk 4]
```diff
@@ -208,47 +210,61 @@ func (p *Plugin) getConnectAccountFirstMessage() string {
 	return fmt.Sprintf(constants.ConnectAccountFirst, fmt.Sprintf(constants.ConnectAccount, p.GetPluginURLPath(), constants.PathOAuthConnect))
 }
 
-func (p *Plugin) ParseSubscriptionsToCommandResponse(subscriptionsList []*serializers.SubscriptionDetails, channelID, createdBy, userID string) string {
+func (p *Plugin) ParseSubscriptionsToCommandResponse(subscriptionsList []*serializers.SubscriptionDetails, channelID, createdBy, userID, command string) string {
 	var sb strings.Builder
 
 	if len(subscriptionsList) == 0 {
-		sb.WriteString(constants.NoSubscriptionFound)
+		sb.WriteString(fmt.Sprintf("No %s subscription exists for this channel", command))
```
Comment (avas27JTG) [inline server/plugin/utils.go:217]:
```suggestion
		sb.WriteString(fmt.Sprintf("No %s subscription exists", command))
```

File: webapp/src/hooks/index.js [hunk 5]
```diff
@@ -32,7 +32,16 @@ export default class Hooks {
 
         if (commandTrimmed && commandTrimmed.startsWith('/azuredevops boards subscription add')) {
             const commandArgs = getCommandArgs(commandTrimmed);
-            this.store.dispatch(setGlobalModalState({modalId: 'subscribeProject', commandArgs}));
+            this.store.dispatch(setGlobalModalState({modalId: 'subscribeProject', commandArgs: [...commandArgs, 'boards']}));
```
Comment (avas27JTG) [inline webapp/src/hooks/index.js:35]: use constant for `boards/repos`

**Comment by avas27JTG** [comment]
@raghavaggarwal2308 also fix the behaviour that we discussed on call

**Review by avas27JTG — APPROVED** [review]

**Comment by avas27JTG** [comment]
@raghavaggarwal2308 
image
